### PR TITLE
Issue #73: remove react internals

### PR DESCRIPTION
**Pull request by alexeybondarenko**
#74

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "react": "0.14.x || 15.x.x"
   },
   "devDependencies": {
+    "babel-eslint": "^7.1.0",
```
Comment (iamdustan) [inline package.json]: why is this in this change?

File: src/index.js [hunk 2]
```diff
@@ -14,6 +12,51 @@ function isRedo(e) {
   return (e.ctrlKey || e.metaKey) && e.keyCode === (e.shiftKey ? KEYCODE_Z : KEYCODE_Y)
 }
 
+function getSelection (el) {
+  var start, end, rangeEl, clone;
+
+  if( el.selectionStart !== undefined){
```
Comment (iamdustan) [inline src/index.js]: nit: `if (typeof el.selectionStart !== 'undefined') {` (spacing should follow the rest of the codebase)

File: src/index.js [hunk 3]
```diff
@@ -14,6 +12,51 @@ function isRedo(e) {
...
+  }
+
+  return { start, end }
+}
+
+function setSelection(el, selection){
+  var rangeEl;
+
+  try {
+    if( el.selectionStart !== undefined){
```
Comment (iamdustan) [inline src/index.js]: nit spacing

File: src/index.js [hunk 4]
```diff
@@ -14,6 +12,51 @@ function isRedo(e) {
...
+    start = el.selectionStart
+    end = el.selectionEnd
+  }
+  else {
+    try {
+      el.focus()
+      rangeEl = el.createTextRange()
+      clone = rangeEl.duplicate()
+
+      rangeEl.moveToBookmark(document.selection.createRange().getBookmark());
```
Comment (iamdustan) [inline src/index.js]: nit: no semicolons

**Comment by iamdustan** [comment]
some nits to get the codestyle in line. the change looks okay to me, though I’d like @jquense to approve, too. He knows the DOM specifics of inputs _far_ better than I!

File: package.json (hunk 1 above)
Comment (alexeybondarenko) [reply package.json]: because Eslint is configured to use babel-eslint as parser and many IDE (such as Atom) require this package to be installed locally.

**Comment by jquense** [comment]
this code has worked fine for me for a long time now. The one caveat is that it aggressively focuses the input (otherwise it'll throw in old IE), before fetching or setting the selection. It shouldn't be a problem but i'd run and confirm it's not changing accidentally conflicting with any other code in the input trying to focus.

File: package.json (hunk 1 above)
Comment (alexeybondarenko) [reply package.json]: I can move this change to separate PR

**Comment by alexeybondarenko** [comment]
I've fixed code style.

**Pull request merged**